### PR TITLE
chore: Convert identities views TestCase to normal test function part 2

### PR DESCRIPTION
**Pull request by zachaysan**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

I'm working through re-writing the TestCase classes into normal test functions. Rewrote the existing TestCase and replaced with fixtures instead of recreating everything. There are other PRs that are similar to this one.

This file was exceptionally large, so I decided to split it up into two different PRs. This PR is the second part of this one: https://github.com/Flagsmith/flagsmith/pull/3519
## How did you test this code?

I manually changed the tests one at a time to refactor them as I went then ran the full test suite afterwards.

### CODE REVIEW

File: api/tests/unit/environments/identities/test_unit_identities_views.py [hunk 1]
```diff
@@ -525,396 +525,439 @@ def test_identities_endpoint_returns_value_for_segment_if_identity_in_segment_an
...
+    environment.save()
+
+    # When
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)
+    response = api_client.post(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
```
Comment (gagantrivedi) [inline api/tests/unit/environments/identities/test_unit_identities_views.py:857]: We should probably add a check to make sure the new traits don't exists

File: api/tests/unit/environments/identities/test_unit_identities_views.py [hunk 2]
```diff
@@ -525,396 +525,439 @@ def test_identities_endpoint_returns_value_for_segment_if_identity_in_segment_an
...
+    environment.allow_client_traits = False
+    environment.save()
+    # When
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment_api_key.key)
+    response = api_client.post(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
```
Comment (gagantrivedi) [inline api/tests/unit/environments/identities/test_unit_identities_views.py:885]: Add a check to make sure traits exists

**Review by gagantrivedi — APPROVED** [review]
One minor comment

File: api/tests/unit/environments/identities/test_unit_identities_views.py (hunk 1 above)
Comment (zachaysan) [reply api/tests/unit/environments/identities/test_unit_identities_views.py:857]: Good idea! Added.

File: api/tests/unit/environments/identities/test_unit_identities_views.py (hunk 2 above)
Comment (zachaysan) [reply api/tests/unit/environments/identities/test_unit_identities_views.py:885]: Added this also.

**Pull request merged**